### ISSUE TITLE
Crash when a node has already ID but gateway doesn't know it

### ISSUE THREAD

**Issue by doudz** [theolind/pymysensors#10]
The lib crash when a node has already and ID but the gateway/controller has been reset.
I modify  _handle_presentation like this  :
```
def _handle_presentation(self, msg):
        """ Processes a presentation message. """
        if msg.child_id == 255 or msg.node_id not in self.sensors :
            # this is a presentation of the sensor platform
            self.add_sensor(msg.node_id)
            self.sensors[msg.node_id].type = msg.sub_type
            self.sensors[msg.node_id].version = msg.payload
            self.alert(msg.node_id)
        if msg.child_id != 255:
            # this is a presentation of a child sensor
            self.sensors[msg.node_id].add_child_sensor(msg.child_id,
                                                       msg.sub_type)
            self.alert(msg.node_id)
```

**Comment by MartinHjelmare** [comment]
Hi!

Setting Sensor.type to msg.sub_type and Sensor.version to msg.payload when msg.child_id is not 255 will make those attributes contain the wrong information. One might argue that those attributes are not that important. You should be aware of this possible problem though.

I would make a third case, which IMO is better, but will still create a situation with partial data for the Sensor attributes.
```
def _handle_presentation(self, msg):
        """ Processes a presentation message. """
        if msg.child_id == 255:
            # this is a presentation of the sensor platform
            self.add_sensor(msg.node_id)
            self.sensors[msg.node_id].type = msg.sub_type
            self.sensors[msg.node_id].version = msg.payload
            self.alert(msg.node_id)
        elif msg.child_id != 255 and msg.node_id not in self.sensors:
            self.add_sensor(msg.node_id)
            self.alert(msg.node_id)
        if msg.child_id != 255:
            # this is a presentation of a child sensor
            self.sensors[msg.node_id].add_child_sensor(msg.child_id,
                                                       msg.sub_type)
            self.alert(msg.node_id)
```

The best way, I think, is to implement handling of *ack* in this library, and use *ack* in your node sketch in the setup function for all the presentation messages etc. That way you can resend the setup messages from the node to the gateway until the node has gotten acknowledgment that everything is setup correctly on the gateway side.

**Comment by oeysteinhansen** [comment]
I tested the code from @MartinHjelmare comment. This worked for me togheter with Home-Assistent

**Comment by MartinHjelmare** [comment]
The crash should be avoided with the fix in #38.

I'm also working on adding support for requesting a new presentation, a feature which is available in mysensors 2.0.
